compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/cyclefind/GlobalVariableRefAnalyzer.java

```diff
@@ -194,7 +196,7 @@ private boolean isFunction(BSymbol symbol) {
         return (symbol.tag & SymTag.FUNCTION) == SymTag.FUNCTION;
     }
 
-    private boolean isGlobalVarSymbol(BSymbol symbol) {
+    private boolean isGlobalVarSymbol(BSymbol symbol, List<BLangVariable> globalVars) {
```

Comment:
Instead of iterating over a list, shall we pass a Set<BSymbol>s and if all global var symbols are included in that set we might be able to remove rest of the checks from this function too.

Reply:
Change to sets in 5d1eea3